### PR TITLE
Update Model Managers

### PR DESCRIPTION
**Pull request by eladprager**
* update model managers: update ModelTrainingManagerInfo, ModelTrainingManager, MultipleDataset, ModelTrainingManager objects
* update model's accuracy: multiply by 100 (will be shown now as 90%).
* update test: update TestMultipleDatasetModelTrainingManager to fit changes

### CODE REVIEW

File: src/pipeline/model/model_trainining_manager.py [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
+    def __init__(self, preprocessor: IPreprocessor, dataset: Dataset, model: IModel):
+        self.preprocessor: IPreprocessor = preprocessor
+        self.training_dataset: Dataset = dataset
+        self.model: IModel = model
 
 
 class ModelTrainingManager(IManager):
-    pass
+    def __init__(self, info: ModelTrainingManagerInfo):
+        self.info = info
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:18]: let's use a property for info, both the getter and setter are needed

File: src/pipeline/model/model_trainining_manager.py [hunk 2]
```diff
@@ -1,30 +1,36 @@
...
 
+    def manage(self):
+        processed_dataset, processed_dataset_plus, feature_metrics_list = self.info.preprocessor.preprocess(self.info.training_dataset)
+        X_train, X_validation, X_test, y_train, y_validation, y_test = self.info.preprocessor.split(processed_dataset)
+        self.info.model.train(X_train, y_train)
+        self.info.model.tune_hyperparameters(X_validation, y_validation)
+        self.info.model.evaluate(X_test, y_test)
+        self.info.model.load(self.info.model.__class__.__name__)
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:26]:
```suggestion
        self._info.model.train(X_train, y_train)
        self._info.model.tune_hyperparameters(X_validation, y_validation)
        self._info.model.evaluate(X_test, y_test)
        self._info.model.load(self.info.model.__class__.__name__)
```

File: src/pipeline/model/model_trainining_manager.py [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
+        return self.info.model
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:27]: In addition, maybe a good idea is to return the feature metrics list too. You can return both the info and the feature metrics list.

File: src/pipeline/model/model_trainining_manager.py [hunk 4]
```diff
@@ -1,30 +1,36 @@
...
+        self.training_dataset: Dataset = dataset
+        self.model: IModel = model
 
 
 class ModelTrainingManager(IManager):
-    pass
+    def __init__(self, info: ModelTrainingManagerInfo):
+        self.info = info
 
+    def manage(self):
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:20]:
```suggestion
    def manage(self) -> Tuple[ModelTrainingManagerInfo, List[IFeatureMetrics]]
```

File: src/pipeline/model/model_trainining_manager.py [hunk 5]
```diff
@@ -1,30 +1,36 @@
...
 
-class MultipleDatasetModelTrainingManager(IManager):
-    def __init__(self, preprocessor: IPreprocessor, dataset: Dataset, model: IModel):
 
-        self.preprocessor = preprocessor
-        self.dataset = dataset
-        self.model = model
+class MultipleDatasetModelTrainingManager(IManager):
+    def __init__(self, info_list: List[ModelTrainingManagerInfo]):
+        self.model_training_managers: List[ModelTrainingManager] = [ModelTrainingManager(info) for info in info_list]
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:32]:
```suggestion
        self._model_training_managers: List[ModelTrainingManager] = [ModelTrainingManager(info) for info in info_list]
```
should be private attribute

File: src/pipeline/model/model_trainining_manager.py [hunk 6]
```diff
@@ -1,30 +1,36 @@
...
-    def manage(self):
-        processed_dataset, processed_dataset_plus, feature_metrics_list = self.preprocessor.preprocess(self.dataset)
-        X_train, X_validation, X_test, y_train, y_validation, y_test = self.preprocessor.split(processed_dataset)
-        self.model.train(X_train, y_train)
-        self.model.tune_hyperparameters(X_validation, y_validation)
-        self.model.evaluate(X_test, y_test)
-        self.model.load(self.model.__class__.__name__)
+    def manage(self) -> List[IModel]:
+        models: List[IModel] = [manager.manage() for manager in self.model_training_managers]
+        return models
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:36]:
```suggestion
        model_training_manger_info_list: List[ModelTrainingManagerInfo] = []
        model_training_manager_feature_metrics_lists: List[List[IFeatureMetric]] = []
        for model_training_manager in self._model_training_managers:
                info, feature_metrics_list = model_training_manager.manage()
                model_training_manger_info_list.append(info)
                model_training_manager_feature_metrics_lists.append(feature_metrics_list)
        
        return model_training_manger_info_list, model_training_manager_feature_metrics_lists
```

File: src/pipeline/model/model_trainining_manager.py [hunk 7]
```diff
@@ -1,30 +1,36 @@
...
+        self.model_training_managers: List[ModelTrainingManager] = [ModelTrainingManager(info) for info in info_list]
 
-    def manage(self):
-        processed_dataset, processed_dataset_plus, feature_metrics_list = self.preprocessor.preprocess(self.dataset)
-        X_train, X_validation, X_test, y_train, y_validation, y_test = self.preprocessor.split(processed_dataset)
-        self.model.train(X_train, y_train)
-        self.model.tune_hyperparameters(X_validation, y_validation)
-        self.model.evaluate(X_test, y_test)
-        self.model.load(self.model.__class__.__name__)
+    def manage(self) -> List[IModel]:
```
Comment (guyfreund) [inline src/pipeline/model/model_trainining_manager.py:34]:
```suggestion
    def manage(self) -> Tuple[List[ModelTrainingManagerInfo], List[List[IFeatureMetrics]]]:
```

**Review by guyfreund — CHANGES_REQUESTED** [review]

**Review by guyfreund — CHANGES_REQUESTED** [review]
Need also to consider returning from both the regular and the multiple managers the evaluation result, i.e. a list of IMetric for each regular manager and a list of lists of IMetric for the multiple manager.

**Review by guyfreund — APPROVED** [review]
